src/main/scala/mesosphere/marathon/core/task/termination/TaskKillService.scala

```diff
@@ -0,0 +1,46 @@
+package mesosphere.marathon.core.task.termination
+
+import mesosphere.marathon.core.task.Task
+
+import scala.concurrent.Future
+
+/**
+  * A service that handles killing tasks. This will take care about extra logic for lost tasks,
+  * apply a retry strategy and throttle kill requests to Mesos.
+  */
+trait TaskKillService {
+  // TODO: should we favor distinct vs overloaded functions? killTask, killTaskById, killTasks, killUnknownTask?
+
+  /**
+    * Kill the given tasks and return a future that is completed when all of the tasks
+    * have been reported as terminal.
+    *
+    * @param tasks the tasks that shall be killed.
+    * @return a future that is completed when all tasks are killed.
+    */
+  def kill(tasks: Iterable[Task]): Future[Unit]
```

Comment:
Nice, I seemed to have missed the introduction of `Done`. Changed it, but the result will be mapped to Unit in places I didn't want to change now.